### PR TITLE
modifications to fix problems with latest numpy version

### PR DESCRIPTION
**Pull request by jeremy9959**
Dear Claudio,

I found your description of word2vec and your implementation of it very useful.  I have made some small changes to the code so that it runs under numpy v 1.14.  I think/hope I've done it correctly, and my version passes your unittests.

The problems all revolved around the intepretation of 1-dimensional arrays in numpy, which are always confusing to me. 

I hope you find this useful.

### CODE REVIEW

File: word2veclite/test/test_word2veclite.py [hunk 1]
```diff
@@ -1,3 +1,4 @@
+import unittest
```
Comment (cbellei) [inline word2veclite/test/test_word2veclite.py:1]: Is this import needed?

File: sample.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+from word2veclite import Word2Vec
+
+corpus = "I like playing football with my friends"
+cbow = Word2Vec(method="skipgram", corpus=corpus,
```
Comment (cbellei) [inline sample.py]: Should be method="cbow" here

**Review by cbellei — CHANGES_REQUESTED** [review]
Hi Jeremy,

Thanks a lot for the PR! This is going to make life easier for several people that had problems with the dependency on numpy so I am looking forward to merge your branch.

I have made a couple of minor inline comments to the code, but the main thing is that the test for the skipgram model throws an error. Also if you try to run the skipgram model with the example in the blogpost
```
corpus = "I like playing football with my friends"
skipgram = Word2Vec(method="skipgram", 
                corpus=corpus,
                window_size=1, n_hidden=2,
                n_epochs=600, learning_rate=0.1)
W1, W2, loss_vs_epoch = skipgram.run()
```
it gives a result for the loss_vs_epoch that is not the same as it was before (see this figure http://www.claudiobellei.com/2018/01/07/backprop-word2vec-python/loss_vs_epoch_skipgram.png)

File: word2veclite/test/test_word2veclite.py (hunk 1 above)
Comment (jeremy9959) [reply word2veclite/test/test_word2veclite.py:1]: WIthout it, here's what happens:
```shell
woodmont ~/GitHub/word2veclite/word2veclite/test > python test_word2veclite.py
Using TensorFlow backend.
Traceback (most recent call last):
  File "test_word2veclite.py", line 90, in <module>
    unittest.main()
NameError: name 'unittest' is not defined
```

Comment (cbellei) [reply word2veclite/test/test_word2veclite.py:1]: Ok got it, I was using nosetests to run the tests. I understand now.

**Comment by cbellei** [comment]
Hi Jeremy, I did a couple more checks. Running `python test_word2veclite.py` results in the tests passing, however when I run `nosetests` the skipgram test fails (not sure why yet, didn't have time to investigate further). In any case, as I mentioned before the results of the skipgram model as they are now are not consistent with the previous ones.

**Comment by jeremy9959** [comment]
I naively accepted the fact that (at least for me) the skipgram test passed, so didn't study it as much as I should have.  I will look more closely at it today.

**Comment by jeremy9959** [comment]
I made another pass at this and hopefully things are OK now with some caveats:
Good News:
both skipgram and cbow run and the loss function seems to behave correctly -- you get convergence.
Bad News:
They fail the test code; the results don't match tensorflow to 5 digits.  I'm not sure if this is a real problem; there could be a bug or it could just be that my particular installation of tensorflow doesn't quite give the same numerical results.  If you believe they should match, I can look again.

**Comment by jeremy9959** [comment]
The jupyter notebook shows the loss function graphs.

File: word2veclite/test/test_word2veclite.py [hunk 3]
```diff
@@ -12,9 +12,9 @@ class TestFunctions(TestCase):
     W2 = np.array([[0.2, 0.2, 0.3, 0.4, 0.5, 0.3, 0.2],
                    [0.3, 0., 0.1, 0., 1., 0.1, 0.5]])
     V, N = W1.shape
-    context_words = np.array([[1, 0, 0, 0, 0, 0, 0], [0, 1, 0, 0, 0, 0, 0],
+    context_words = np.matrix([[1, 0, 0, 0, 0, 0, 0], [0, 1, 0, 0, 0, 0, 0],
```
Comment (cbellei) [inline word2veclite/test/test_word2veclite.py]: better to leave this as np.array

File: word2veclite/test/test_word2veclite.py [hunk 4]
```diff
@@ -12,9 +12,9 @@ class TestFunctions(TestCase):
...
                               [0, 0, 0, 1, 0, 0, 0], [0, 0, 0, 0, 1, 0, 0]])
-    center_word = np.array([0., 0., 1., 0., 0., 0., 0.])
+    center_word = np.matrix([0., 0., 1., 0., 0., 0., 0.])
```
Comment (cbellei) [inline word2veclite/test/test_word2veclite.py]: same here, better to leave it as np.array

**Review by cbellei — CHANGES_REQUESTED** [review]
It is best to leave the tensorflow part as is (so avoid changing anything that it depends on, like context_words and center_word).

To make the test work, best to change line 21 of `test_word2veclite.py` into`W1_m, W2_m, loss_m = cbow.cbow(np.asmatrix(self.context_words), np.asmatrix(self.center_word), self.W1, self.W2, 0.)`
and line  22 as `W1_m, W2_m, loss_m = skipgram.skipgram(np.asmatrix(self.context_words), np.asmatrix(self.center_word), self.W1, self.W2, 0.)`.

This ensures that the new codebase is called using matrices, while the tensorflow section stays as is.

**Comment by jeremy9959** [comment]
OK, I made those changes, and now the tests pass.

**Comment by jeremy9959** [comment]
Incidentally I haven't tested the "predict" function nor used a larger corpus or other parameters like a wider window.

**Comment by jeremy9959** [comment]
```bash
woodmont:~/GitHub/word2veclite/word2veclite/test$ nosetests
Using TensorFlow backend.
2018-07-09 17:12:26.126068: I tensorflow/core/platform/cpu_feature_guard.cc:140] Your CPU supports instructions that th\
is TensorFlow binary was not compiled to use: AVX2 FMA
..
----------------------------------------------------------------------
Ran 2 tests in 0.233s

OK
```

**Comment by cbellei** [comment]
Great, I will test this and merge your branch later tonight.

**Comment by jeremy9959** [comment]
Cool! Thanks, and thanks again for putting this all together in the first place, I learned a lot.

**Pull request merged**